src/write.c

```diff
@@ -402,8 +396,23 @@ static uint8_t avifItemPropertyDedupFinish(avifItemPropertyDedup * dedup, avifRW
     property->index = ++dedup->nextIndex; // preincrement so the first new index is 1 (as ipma is 1-indexed)
     property->size = newPropertySize;
     property->offset = avifRWStreamOffset(outputStream);
-    avifRWStreamWrite(outputStream, dedup->buffer.data, newPropertySize);
-    return property->index;
+    *propertyIndex = property->index;
+    AVIF_CHECKRES(avifRWStreamWrite(outputStream, dedup->buffer.data, newPropertySize));
+    return AVIF_RESULT_OK;
+}
+
+static avifResult avifItemPropertyDedupFinishAndIpmaPush(avifItemPropertyDedup * dedup,
```

Comment:
Nit/optional: This can be the new `avifItemPropertyDedupFinish()` function. We don't need two functions.